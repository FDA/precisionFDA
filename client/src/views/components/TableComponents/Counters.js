import React from 'react'
import PropTypes from 'prop-types'

import './style.sass'


const Counters = ({ currentPage, nextPage, totalCount, count }) => {
  const from = nextPage ? (currentPage - 1) * count + 1 : totalCount - count + 1
  const to = nextPage? currentPage * count : totalCount

  return (
    <div className="pfda-table-counters">
      {`${from}-${to}/${totalCount}`}
    </div>
  )
}

Counters.propTypes = {
  currentPage: PropTypes.number,
  nextPage: PropTypes.number,
  totalCount: PropTypes.number,
  count: PropTypes.number,
}

export default Counters
