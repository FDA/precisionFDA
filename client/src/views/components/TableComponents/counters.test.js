import React from 'react'
import { shallow } from 'enzyme'

import Counters from './Counters'


describe('Counters test', () => {
  it('should render', () => {
    const data = {
      currentPage: 2,
      nextPage: 3,
      totalCount: 4,
      count: 5,
    }
    const component = shallow(<Counters data={data} />)
    expect(component).toMatchSnapshot()
  })
})
