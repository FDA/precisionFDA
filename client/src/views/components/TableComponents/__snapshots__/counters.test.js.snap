// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counters test should render 1`] = `
<div
  className="pfda-table-counters"
>
  NaN-undefined/undefined
</div>
`;
